icu4c/source/data/xml/brkitr/de.xml

```diff
@@ -6,9 +6,9 @@
 
  See http://bugs.icu-project.org/trac/ticket/10750#comment:5 - this file is needed to emit break iterator rules properly.
 -->
-<!DOCTYPE ldml SYSTEM "http://www.unicode.org/repos/cldr/trunk/common/dtd/ldml.dtd"
+<!DOCTYPE ldml SYSTEM "https://raw.githubusercontent.com/unicode-org/cldr/master/common/dtd/ldml.dtd"
```

Comment:
Probably should be a specific version X instead of master. Or perhaps, copy version X INTO ICU's source base, and use that. I'm not sure if using this URL is the best practice.

Reply:
I kind of think we should basically "vendor" the CLDR dtd here, only for this use. And change this to a relative link:   `../dtd/cldr-35/ldml.dtd`. And then NOT update that dtd, ever, unless these override files actually depend on something that changes.

Reply:
The status quo is that it points to CLDR SVN Trunk.  So, pointing to CLDR GitHub Master is the equivalent.

If there's a way to avoid the need for an HTTP(S) request when running Ldml2IcuConverter, that would be a desirable outcome.  I'm not sure about how we'd keep the ICU copy of ldml.dtd in sync with the CLDR copy, though.  That file does tend to change from version to version I believe.

Reply:
It's the status quo, but not a good idea.  CLDR used to have HTTP references long ago also and removed them.

>  I'm not sure about how we'd keep the ICU copy of ldml.dtd in sync with the CLDR copy, though. That file does tend to change from version to version I believe.

So here's the thing. Yes, the dtd changes but that may not mean _the version which ICU points to needs to change_.  From a cursory look, most files haven't had a substantive change in many years, and unlikely a change which is affected by the DTD. We can always upgrade the vendored DTD if/when the need arises.

Reply:
It seems like the most "correct" thing to do would be to pull the dtd from the local CLDR clone that is being used when running Ldml2IcuConverter.  Is that something we can do?  Like, could we reference an environment variable like `$CLDR_DIR` in the path?

Reply:
Why is that more correct?

Reply:
Well actually.  I guess, when running Ldml2IcuConverter, the DTD that applies to the files *in the CLDR project* should be the one from the CLDR project.  But, these files live in ICU.  So, if the DTD were to ever change in a breaking way, the ICU files would need to be updated.

The DTD version from which these files were written should be versioned, and saving the DTD file into ICU is a way to version it.

Reply:
Referencing an environment variable like $CLDR_DIR in the path would be ideal for our case, since we, on Android, checkouts both ICU and CLDR directory. But I have no strong opinion here, and "vendor" the CLDR dtd would work for us too.

However, as @sffc said, pointing to CLDR GitHub Master is the equivalent. And the XML file is currently broken because of the CLDR github migration. Can we fix the regression first before any further improvement?

Reply:
CLDR's tooling only checks that the path ends in `common/dtd/ldmlICU.dtd`.  so I propose that right now, we *copy* the file in question to `icu/icu4c/source/data/dtd/cldr-35.1/common/dtd/ldmlICU.dtd` and then change the paths to `../../../dtd/cldr-35.1/common/dtd/ldmlICU.dtd`

then removing the file from CLDR can be done as a separate step.

Reply:
^ and ldml.dtd similarly

Reply:
Ok. I will update this PR to copy the CLDR version 35.1 of the DTD files to icu/icu4c/source/data/dtd/cldr-35.1/common/dtd/

Reply:
It's now done.